### ISSUE TITLE
Rule change proposal: order-in-routes execution order

### ISSUE THREAD

**Issue by rwwagner90** [ember-cli/eslint-plugin-ember#119]
I find it a bit strange that `beforeModel` is placed after `model` in the `order-in-routes` rule. I also think there is probably some confusion in the community around the order the route hooks fire in. Would it not make more sense to order based on execution order of the hooks?

Something like this:

```
beforeModel(){}
model(){}
afterModel(){}
setupController(){}
```

**Comment by kevinkucharczyk** [comment]
@rwwagner90 that sounds like a great idea! You could do the same in `order-in-components` as well.

**Comment by rwwagner90** [comment]
@kevinkucharczyk would that be supported as the official order, if I were to PR it in?

**Comment by kevinkucharczyk** [comment]
@rwwagner90 I think the recommended order should be the hook order execution, though perhaps it would be nice to give users an option to define their own order.

**Comment by jbandura** [comment]
Sounds like a great idea 👍

**Comment by michalsnik** [comment]
Released in `v4.4.0` 🚀 Thanks @rwwagner90
